rspec-expectations/lib/rspec/matchers/built_in/change.rb

```diff
@@ -150,8 +150,8 @@ def matches?(event_proc)
 
         # @private
         def does_not_match?(_event_proc)
-          raise NotImplementedError, "`expect { }.not_to change " \
-                                     "{ }.#{@relativity}()` is not supported"
+          raise NotImplementedError, "Using `expect { }.not_to change { }` " \
+                                     "with the `#{@relativity}()` matcher is not supported."
```

Comment:
Strictly speaking, `by()` is not a matcher. I've seen those were mentioned as "qualifiers". The `change` is the matcher here.


WDYT of "Using a negated form of the `change` matcher with a `to()` qualifier is not supported."?

Reply:
One other case is when the negation happens not with a `not_to`/`to_not`, but rather by creating a negated version with `define_negated_matcher :not_change, :change`.

The suggested error message would be less confusing for a negated matcher.

Reply:
That sounds good to me, it's clearly pointing out the error so I'm happy with that

Reply:
Even “ Using a negated form of the `change` matcher with `to()` is not supported.” is good, without all the too-technical terms involved.

Reply:
yeah I like that, no need for complicated lingo!